tutorials/math/principal.py

```diff
@@ -0,0 +1,88 @@
+## \file
+##    \ingroup tutorial_math
+## \notebook
+## Principal Components Analysis (PCA) example
+##
+## Example of using TPrincipal as a stand alone class.
+##
+## We create n-dimensional data points, where c = trunc(n / 5) + 1
+## are  correlated with the rest n - c randomly distributed variables.
+##
+## \macro_output
+## \macro_code
+##
+## \authors Juan Fernando Jaramillo Botero
+##
+## based on principal.C by Rene Brun and Christian Holm Christensen
+
+from ROOT import TPrincipal, gRandom, TBrowser
+import numpy
+
+
+def principal(n=10, m=10000):
+    c = n / 5 + 1
+
+    print ("*************************************************")
+    print ("*         Principal Component Analysis          *")
+    print ("*                                               *")
+    print ("*  Number of variables:           {0:4d}          *".format(n))
+    print ("*  Number of data points:         {0:8d}      *".format(m))
+    print ("*  Number of dependent variables: {0:4d}          *".format(c))
+    print ("*                                               *")
+    print ("*************************************************")
+
+    # Initilase the TPrincipal object. Use the empty string for the
+    # final argument, if you don't wan't the covariance
+    # matrix. Normalising the covariance matrix is a good idea if your
+    # variables have different orders of magnitude.
+    principal = TPrincipal(n, "ND")
+
+    # Use a pseudo-random number generator
+    randumNum = gRandom
+
+    # Make the m data-points
+    # Make a variable to hold our data
+    # Allocate memory for the data point
+    data = []
```

Comment:
in order to avoid numpy arrays, here you can use the array module or an stl vector
```
data = ROOT.vector('double')()
```
then instead of invoking the *append* method use *push_back*.

Reply:
Done

Reply:
Sorry, I had a problem, but I already solved. I will push the solution.

Reply:
Done, I change append by push_back in the vector.